### ISSUE TITLE
Bigint prevents use on edge environments

### ISSUE THREAD

**Issue by TheoBr** [koskimas/kysely#142]
Tried using kysely in an Astro edge env as a string builder for the new planetscale http client. Was able to work around with a nasty patch package lol. Literally just had to change the lock id for postgres to not be a bigint

<img width="909" alt="image" src="https://user-images.githubusercontent.com/6751787/185560468-ea1bdd68-d256-4205-a7b6-cb317f08072e.png">

Patch:
```
diff --git a/node_modules/kysely/dist/cjs/dialect/postgres/postgres-adapter.js b/node_modules/kysely/dist/cjs/dialect/postgres/postgres-adapter.js
index 08ef143..301315f 100644
--- a/node_modules/kysely/dist/cjs/dialect/postgres/postgres-adapter.js
+++ b/node_modules/kysely/dist/cjs/dialect/postgres/postgres-adapter.js
@@ -4,7 +4,7 @@ exports.PostgresAdapter = void 0;
 const sql_js_1 = require("../../raw-builder/sql.js");
 const dialect_adapter_base_js_1 = require("../dialect-adapter-base.js");
 // Random id for our transaction lock.
-const LOCK_ID = 3853314791062309107n;
+const LOCK_ID = 3853314791062309107;
 class PostgresAdapter extends dialect_adapter_base_js_1.DialectAdapterBase {
     get supportsTransactionalDdl() {
         return true;
diff --git a/node_modules/kysely/dist/esm/dialect/postgres/postgres-adapter.js b/node_modules/kysely/dist/esm/dialect/postgres/postgres-adapter.js
index 2596e72..c48e562 100644
--- a/node_modules/kysely/dist/esm/dialect/postgres/postgres-adapter.js
+++ b/node_modules/kysely/dist/esm/dialect/postgres/postgres-adapter.js
@@ -2,7 +2,7 @@
 import { sql } from '../../raw-builder/sql.js';
 import { DialectAdapterBase } from '../dialect-adapter-base.js';
 // Random id for our transaction lock.
-const LOCK_ID = 3853314791062309107n;
+const LOCK_ID = 3853314791062309107;
 export class PostgresAdapter extends DialectAdapterBase {
     get supportsTransactionalDdl() {
         return true;

```

**Comment by TheoBr** [comment]
Github repo w/ this working btw https://github.com/TheoBr/pscalebench

**Comment by koskimas** [comment]
I think that number gets truncated, as it's greater than the largest integer that can be expressed as a JavaScript number. Does a string work?

**Comment by TheoBr** [comment]
@koskimas Yes string works as well (at least passes compilation - I'm not using the driver)

Not sure about truncation, will bring up w/ the Astro guys as the warning seems to exist in that layer

**Comment by koskimas** [comment]
Kysely does have a minimum node version which does have bigints, but I can fix this particular issue if you can come up with a patch that works.

But kysely can break for those old node versions at any given time. There is no guarantee that things will work under node 14 (or any browse that has a subset of node 14 features implemented).

**Comment by TheoBr** [comment]
To be clear this isn't an old node version - this is edge support (similar to Deno if anything)

**Comment by koskimas** [comment]
Kysely only officially supports node. You need to compare the features to the minimum supported node version, which is 14.

**Comment by jackharrhy** [comment]
Running into this issue while trying to use Astro, not even using an edge environment here, seems to break with Astro's dev server (`npm run dev`)

**Comment by igalklebanov** [comment]
Had similar issues with `n` notation and applied a similar patch a few months ago - node 14, but esbuild configured with a target that doesn't support it.

The following is truthy in chrome, node 16.x, node 14.x, and when changing to lhv, astro server starts:

```
BigInt('3853314791062309107') === 3853314791062309107n
```

**Comment by igalklebanov** [comment]
@t3dotgg Fix released in Kysely `0.23`.
